dvc/lock.py

```diff
@@ -26,10 +27,56 @@ class LockError(DvcException):
     """Thrown when unable to acquire the lock for dvc repo."""
 
 
+class Lock(object):
+    """Class for dvc repo lock.
+
+    Uses zc.lockfile as backend.
+    """
+
+    def __init__(self, lockfile, friendly=False, **kwargs):
+        self._friendly = friendly
+        self.lockfile = lockfile
+        self._lock = None
+
+    @property
+    def files(self):
+        return [self.lockfile]
+
+    def _do_lock(self):
+        try:
+            with Tqdm(
+                bar_format="{desc}",
+                disable=not self._friendly,
+                desc="If dvc froze, see `nfs_lock` in man.dvc.org/config#core",
+            ):
+                self._lock = zc.lockfile.LockFile(self.lockfile)
+        except zc.lockfile.LockError:
+            raise LockError(FAILED_TO_LOCK_MESSAGE)
+
+    def lock(self):
+        try:
+            self._do_lock()
+            return
+        except LockError:
+            time.sleep(DEFAULT_TIMEOUT)
```

Comment:
Some people might think it froze here.

Reply:
@Suor It is an old code though. At least a message like "sleeping for 5 sec" or smth better would be nice here, agreed.

Reply:
It is old, but we didn't have that "freeze message" before. So before people will simply think it's slow, now they will wonder whether it's frozen or just slow.

Reply:
The message won't be seen, as we are not sleeping with that message on. So the operation is the same as it was before.